### PR TITLE
[FLINK-12820][Connectors / Cassandra] Support ignoring writing nulls 

### PR DESCRIPTION
**Pull request by ozancicek**
## What is the purpose of the change
Added support to ignore null fields when writing to c* for tuple types. 

## Brief change log
 - Added 'ignoreNullFields' config parameter which can be enabled
 - Unset nulls from bound statement.

## Verifying this change

This change added tests and can be verified as follows:
 Added a test case to CassandraConnectorITCase testing partial column updates

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (JavaDocs)

### CODE REVIEW

File: flink-connectors/flink-connector-cassandra/src/main/java/org/apache/flink/streaming/connectors/cassandra/CassandraSinkBaseConfig.java [hunk 1]
```diff
@@ -38,6 +38,12 @@
 	 */
 	public static final Duration DEFAULT_MAX_CONCURRENT_REQUESTS_TIMEOUT = Duration.ofMillis(Long.MAX_VALUE);
 
+	/**
+	 * The default option to ignore null fields on insert. By default, {@code Boolean.FALSE}.
```
Comment (Myasuka) [inline flink-connectors/flink-connector-cassandra/src/main/java/org/apache/flink/streaming/connectors/cassandra/CassandraSinkBaseConfig.java]: should be `on insertion`

Comment (ozancicek) [reply flink-connectors/flink-connector-cassandra/src/main/java/org/apache/flink/streaming/connectors/cassandra/CassandraSinkBaseConfig.java]: Thanks, fixed!

**Comment by StephanEwen** [comment]
@flinkbot approve-until architecture

**Comment by StephanEwen** [comment]
Overall approach looks good.

I would ask you to switch to use primitive booleans rather than boxed booleans, to be consistent with the approach in the remainder of Flink. This also avoid accidental nullability and late null pointer exceptions.

Otherwise looks good to me.

**Comment by StephanEwen** [comment]
@flinkbot approve all

**Comment by StephanEwen** [comment]
Looks good to me, merging this...

**Pull request closed without merging**